app/src/main/java/com/duckduckgo/app/pixels/ppropromo/params/AdditionalPixelParamPlugin.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.pixels.ppropromo.params
+
+import com.duckduckgo.anvil.annotations.ContributesPluginPoint
+import com.duckduckgo.di.scopes.AppScope
+
+/**
+ * A plugin point for additional pixel parameters that are allowed to be appended as additional parameters
+ */
+@ContributesPluginPoint(AppScope::class)
```

Comment:
Initially added this to common-utils with different implementations spread out across different impl modules.

However, since the consumer of AdditionalPixelParamPlugin is not “publicly” available yet (not a real API), I just moved everything to : app module.

Reply:
If we want to make this an API, we can structure it like this:
- `AdditionalPixelParamPlugin` > common-utils
- `PixelParamsAdditionPlugin` > common-utils
- `PixelParamsAdditionInterceptor` > :app
- Feature flag + rc > :app
- `AdditionalPixelParamsGenerator` > :app
- Impl of `AdditionalPixelParamPlugin` > any module
- Impl of `PixelParamsAdditionPlugin` > any module

But I held off for now since, the feature flag is quite specific to ppro needs (`origin` eligibility).